python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -159,6 +159,86 @@ def condition():
         ray.get_actor("abc")
 
 
+@pytest.mark.parametrize(
+    "ray_start_cluster_head", [
+        generate_system_config_map(
+            num_heartbeats_timeout=20, ping_gcs_rpc_server_max_retries=60)
+    ],
+    indirect=True)
+def test_create_pg_after_gcs_server_restart(ray_start_cluster_head):
+    cluster = ray_start_cluster_head
+    cluster.add_node(num_cpus=4)
+    cluster.add_node(num_cpus=4)
+    cluster.wait_for_nodes()
+
+    # Create placement group 1 successfully.
+    placement_group1 = ray.util.placement_group([{"CPU": 2}, {"CPU": 2}])
+    ray.get(placement_group1.ready(), timeout=2)
+
+    # Restart gcs server.
+    cluster.head_node.kill_gcs_server()
+    cluster.head_node.start_gcs_server()
+
+    # Create placement group 2 successfully.
+    placement_group2 = ray.util.placement_group([{"CPU": 2}, {"CPU": 2}])
+    ray.get(placement_group2.ready(), timeout=2)
+
+    # Create placement group 3.
+    # Status is `PENDING` because the cluster resource is insufficient.
+    placement_group3 = ray.util.placement_group([{"CPU": 2}, {"CPU": 2}])
+    with pytest.raises(ray.exceptions.GetTimeoutError):
+        ray.get(placement_group3.ready(), timeout=2)
+    table = ray.util.placement_group_table(placement_group3)
+    assert table["state"] == "PENDING"
+
+
+@pytest.mark.parametrize(
+    "ray_start_cluster_head", [
+        generate_system_config_map(
+            num_heartbeats_timeout=20, ping_gcs_rpc_server_max_retries=60)
+    ],
+    indirect=True)
+def test_create_actor_with_pg_after_gcs_server_restart(ray_start_cluster_head):
+    cluster = ray_start_cluster_head
+    cluster.add_node(num_cpus=4)
+    cluster.wait_for_nodes()
+
+    # Create a placement group.
+    placement_group = ray.util.placement_group([{"CPU": 2}, {"CPU": 2}])
+
+    # Create an actor that occupies resources after gcs server restart.
+    cluster.head_node.kill_gcs_server()
+    cluster.head_node.start_gcs_server()
+    actor_2 = Increase.options(
+        placement_group=placement_group,
+        placement_group_bundle_index=1).remote()
+    assert ray.get(actor_2.method.remote(1)) == 3
+
+
+@pytest.mark.parametrize(
+    "ray_start_cluster_head", [
+        generate_system_config_map(
+            num_heartbeats_timeout=20, ping_gcs_rpc_server_max_retries=60)
+    ],
+    indirect=True)
+def test_create_pg_during_gcs_server_restart(ray_start_cluster_head):
```

Comment:
I wonder if we can add more chaotic and undeterministic tests. For example, in some loop, creating pgs, waiting until it is scheduled with 50% probably, and kill gcs server with other 50% probability or something like that. Do you think it is okay to add here? (or should it be a release test? )

Reply:
Fixed, thanks.